internal/mesh/internal/controllers/implicitdestinations/auth_helper_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package implicitdestinations
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	rtest "github.com/hashicorp/consul/internal/resource/resourcetest"
+	pbauth "github.com/hashicorp/consul/proto-public/pbauth/v2beta1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+// TODO: do this properly and export it from internal/auth/exports.go
+// This is a crude approximation suitable for this test.
+func ReconcileComputedTrafficPermissions(
```

Comment:
@skpratt 
Head's up that at some point in the future it may be useful to export a portion of the interior of the CTP controller as a method in `internal/auth/trafficpermtest` for use outside of the package to create valid CTPs for other controller tests. I did a reduced version of that here for my own purposes, but having a more official one would be nice.

I found that was useful to properly create a test ComputedRoutes resource.